editoast/src/views/work_schedules.rs

```diff
@@ -275,6 +295,238 @@ async fn project_path(
+    )
+)]
+async fn create_group(
+    State(app_state): State<AppState>,
+    Extension(auth): AuthenticationExt,
+    Json(WorkScheduleGroupCreateForm {
+        work_schedule_group_name,
+    }): Json<WorkScheduleGroupCreateForm>,
+) -> Result<Json<WorkScheduleGroupCreateResponse>> {
+    let authorized = auth
+        .check_roles([BuiltinRole::WorkScheduleWrite].into())
+        .await
+        .map_err(AuthorizationError::AuthError)?;
+    if !authorized {
+        return Err(AuthorizationError::Unauthorized.into());
+    }
+
+    let db_pool = app_state.db_pool_v2.clone();
+    let conn = &mut db_pool.get().await?;
+    let group_name = work_schedule_group_name.unwrap_or(Uuid::new_v4().to_string());
+
+    // Create the work_schedule_group
+    let work_schedule_group = WorkScheduleGroup::changeset()
+        .name(group_name.clone())
+        .creation_date(Utc::now())
+        .create(conn)
+        .await;
+    let work_schedule_group = work_schedule_group.map_err(|e| map_diesel_error(e, group_name))?;
+    Ok(Json(WorkScheduleGroupCreateResponse {
+        work_schedule_group_id: work_schedule_group.id,
+    }))
+}
+
+#[utoipa::path(
+    delete, path = "",
+    tag = "work_schedules",
+    params(WorkScheduleGroupIdParam),
+    responses(
+        (status = 204, description = "The work schedule group has been deleted"),
+        (status = 404, description = "The work schedule group does not exist"),
+    )
+)]
+async fn delete_group(
+    State(app_state): State<AppState>,
+    Extension(auth): AuthenticationExt,
+    raw_group_id: Path<WorkScheduleGroupIdParam>,
+) -> Result<impl IntoResponse> {
+    let authorized = auth
+        .check_roles([BuiltinRole::WorkScheduleWrite].into())
+        .await
+        .map_err(AuthorizationError::AuthError)?;
+    if !authorized {
+        return Err(AuthorizationError::Unauthorized.into());
+    }
+
+    let db_pool = app_state.db_pool_v2.clone();
+    let conn = &mut db_pool.get().await?;
+    let group_id = raw_group_id.id;
+
+    conn.transaction(|conn| {
+        Box::pin(async move {
+            // Check that the group exists
+            let work_schedule_group =
+                WorkScheduleGroup::retrieve_or_fail(&mut conn.clone(), group_id, || {
+                    WorkScheduleError::WorkScheduleGroupNotFound { id: group_id }
+                })
+                .await?;
+
+            // List work schedules to be deleted
+            let selection_setting = SelectionSettings::new()
+                .filter(move || WorkSchedule::WORK_SCHEDULE_GROUP_ID.eq(group_id));
+            let work_schedule_ids = WorkSchedule::list(&mut conn.clone(), selection_setting)
+                .await?
+                .into_iter()
+                .map(|work_schedule| work_schedule.id);
+
+            // Delete them
+            use crate::models::DeleteBatch;
+            let conn = &mut db_pool.get().await?;
+            WorkSchedule::delete_batch(conn, work_schedule_ids).await?;
```

Comment:
They might be automatically deleted along with their group ? If so, that `delete_batch` of the work schedules can be skipped

Reply:
I'm not sure (it's the first time I really deal with editoast tables).

From what I can see, the `work_schedule_group_id` is just an `i64` without any magic or macro to indicate that it's a foreign key, I don't think the database knows about this link. But that sounds like troubles, maybe there's something missing.

I don't know, @leovalais I'd need the input of someone more familiar

Reply:
You can check constraints and foreign keys in the `up.sql` script that generates the table in `migrations` and get some insight about the tables in `tables.rs`. From what I see in the sql file, there is a foreign key with `ON DELETE CASCADE` between work schedules and their group, so only deleting the group should be fine

Reply:
Checking the `up.sql` script to get some info about the tables is what I find to be the most readable but if you do, make sure to grep that there isn't an `ALTER TABLE` later on in the migrations that impacts what you are looking for

Reply:
Fixed in https://github.com/OpenRailAssociation/osrd/pull/9785/commits/fadd260d6bd8557a2c7764ed98357c71ca12787e